src/components/NavBar/index.tsx

```diff
@@ -0,0 +1,108 @@
+'use client';
+
+import { useMemo } from 'react';
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+import Icon from '@/components/Icon';
+import COLORS from '@/styles/colors';
+import { Flex } from '@/styles/containers';
+import { useProfile } from '@/utils/ProfileProvider';
+import { ProfileButton, SmallLinkButton } from '../Buttons';
+import * as Styles from './style';
+
+export default function NavBar() {
+  const profile = useProfile();
+  const AuthButtonView = useMemo(() => {
+    if (!profile) throw new Error('Profile must be defined.');
+    if (profile.profileReady && profile?.profileData)
+      return (
+        <ProfileButton href="/settings">
+          {profile.profileData?.first_name || 'Profile'}
+        </ProfileButton>
+      );
+
+    return (
+      <>
+        <SmallLinkButton
+          $primaryColor={COLORS.blueMid}
+          $secondaryColor={COLORS.blueDark}
+          href="/login"
+        >
+          Log In
+        </SmallLinkButton>
+        <SmallLinkButton
+          $primaryColor="white"
+          $secondaryColor={COLORS.blueDark}
+          href="/signup"
+          style={{ color: COLORS.blueMid }}
+        >
+          Sign Up
+        </SmallLinkButton>
+      </>
+    );
+  }, [profile]);
+
+  function useActiveStatus(path: string): boolean {
+    const currentPath = usePathname();
+    return currentPath.includes(path);
+  }
+
+  type NavLink = {
+    name: string;
+    path: string;
+    active: boolean;
+  };
+  const navlinks: NavLink[] = [
+    { name: 'Cases', path: '/cases', active: useActiveStatus('/cases') },
+    {
+      name: 'Limited Case Assignments',
+      path: '/limited-case-assignments',
+      active: useActiveStatus('/limited-case-assignments'),
+    },
+    {
+      name: 'Language Support',
+      path: '/language-support',
+      active: useActiveStatus('/language-support'),
+    },
+  ];
+
+  const renderLink = (link: NavLink) => (
+    <Styles.LinkContainer>
+      <Flex
+        $direction="column"
+        $justify="center"
+        style={{
+          flex: 1,
+        }}
+      >
+        <Styles.NoUnderlineLink
+          href={link.path}
+          $color="white"
+          $isActive={link.active}
+        >
+          {link.name}
+        </Styles.NoUnderlineLink>
```

Comment:
unfortunately, the changing font weights will cause the layout of the navbar to shift as bolded text take up more space

to remedy this, we can either fix the width of each box, or try a more...
<details>
<summary>unorthodoxed approach</summary>

- make a duplicate of this text with the same font styles
- group the two in a box with position relative
- make one of the text absolute positioned
    - call the absolute positioned text "display text"
    - call the other one "layout text"
- make sure the two texts line up and share the same center
    - this can be achieved by giving the display text the following styles:
        - `top: 50%`
        - `left: 50%`
        - `transform: translate(-50%, -50%)`
    - and centering the layout text as normal
- make the layout text bolded, always
- give the display text the conditional bolding (i.e. with `$isActive`)
</details>